src/main/java/Duke.java

```diff
@@ -1,10 +1,159 @@
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
 public class Duke {
+    public static String h_rule = "────────────────────────────────────────────────────────────────────";
+
+    public static int dukeRunner(String log) throws DukeNotFoundException, DukeDescriptionException, DukeTimingException, FileNotFoundException {
+        Scanner sc = new Scanner(System.in);
+
+        String raw_in;
+        String[] input;
+        List<Task> itemList = new ArrayList<>();
+        int count = 0;
+        List<String> fileContents = null;
+
+        try {
```

Comment:
Perhaps all the file-based IO can be abstracted away to another class for easier setup/teardown and CRUD operations. Is there a reason to keep it purely within the main class?